scrcpy/src/main/java/io/github/miuzarte/scrcpyforandroid/NativeCoreFacade.kt

```
@@ -0,0 +1,833 @@
+    }
+
+    /**
+     * Execute a blocking IO task on the internal single-thread executor and return its result.
+     *
+     * - This serializes access to native adb/session operations and unwraps
+     *   ExecutionException to rethrow the underlying cause.
+     */
+    private fun <T> ioCall(task: () -> T): T {
+        return try {
+            executor.submit<T> { task() }.get()
+        } catch (e: ExecutionException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)